### PR TITLE
BC-3847 - update board element

### PR DESCRIPTION
**Pull request by hoeppner-dataport**
# Short Description
Endpoint for updating different types of board-elements.

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-3847
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4103

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/feature-board/state/ContentElementState.composable.ts [hunk 1]
```diff
@@ -37,8 +41,9 @@ export const useContentElementState = <T extends AnyContentElement>(
 		}
 	);
 
-	const updateElement = (payload: T["content"]) => {
+	const updateElement = async (payload: T["content"]) => {
 		console.log("update element", { ...payload });
```
Comment (NFriedo) [inline src/components/feature-board/state/ContentElementState.composable.ts:45]: Is this console.log intended?

Comment (hoeppner-dataport) [reply src/components/feature-board/state/ContentElementState.composable.ts:45]: It is intended to keep keeping it :-)

File: src/components/feature-board/state/ContentElementState.composable.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
+import { TextElementContent } from "@/serverApi/v3";
```
Comment (NFriedo) [inline src/components/feature-board/state/ContentElementState.composable.ts:1]: TextContentElement is not used :)

File: src/components/feature-board/state/ContentElementState.composable.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
 import { watchDebounced } from "@vueuse/core";
 import { ref, toRef, unref, watch } from "vue";
+import { useBoardApi } from "../shared/BoardApi.composable";
 import { useInlineEditInteractionHandler } from "../shared/InlineEditInteractionHandler.composable";
-import { AnyContentElement } from "../types/ContentElement";
+import { AnyContentElement, TextContentElement } from "../types/ContentElement";
```
Comment (NFriedo) [inline src/components/feature-board/state/ContentElementState.composable.ts:6]: This TextContentElement is also not used

Comment (hoeppner-dataport) [reply src/components/feature-board/state/ContentElementState.composable.ts]: fixed

**Review by OliverHappe — APPROVED** [review]

**Review by muratmerdoglu-dp — DISMISSED** [review]

**Review by muratmerdoglu-dp — APPROVED** [review]

**Pull request merged**